tests/tt_metal/tt_metal/unit_tests_fast_dispatch_single_chip_multi_queue/command_queue/test_EnqueueTrace.cpp

```diff
@@ -94,20 +89,160 @@ TEST_F(MultiCommandQueueSingleDeviceFixture, EnqueueOneProgramTrace) {
         input_data[i] = i;
     }
 
-    EnqueueWriteBuffer(data_movement_queue, input, input_data.data(), true);
-    Trace trace = BeginTrace(trace_queue);
-    EnqueueProgram(trace_queue, simple_program, false, trace);
-    Finish(trace_queue);
-    EndTrace(trace);
+    // Eager mode
     vector<uint32_t> eager_output_data;
     eager_output_data.resize(input_data.size());
+
+    EnqueueWriteBuffer(data_movement_queue, input, input_data.data(), true);
+    EnqueueProgram(command_queue, simple_program, true);
     EnqueueReadBuffer(data_movement_queue, output, eager_output_data.data(), true);
 
+    // Trace mode
     vector<uint32_t> trace_output_data;
     trace_output_data.resize(input_data.size());
-    EnqueueTrace(trace, true);
+
+    Trace trace;
+    EnqueueWriteBuffer(data_movement_queue, input, input_data.data(), true);
+
+    BeginTrace(trace);
+    EnqueueProgram(trace.trace_queue(), simple_program, false);
+    EndTrace(trace);
+    // Instantiate a trace on a device queue
+    uint32_t trace_id = InstantiateTrace(trace, command_queue);
```

Comment:
I'm looking at the code, and I'm thinking this may be a bit dangerous. I was thinking that the trace has a move constructor and gets assigned to the command queue. But it looks like we just assign a ptr to the command queue. 
This means if the Trace object goes out of scope, the cq's trace_ptr will be stale.

I think it may be best for this API to give cq full ownership of the trace (maybe I'm missing something).

Reply:
This is temporary for implementing the stub functionality, the actual instantiate should not need the trace ptr assignment into the CQ, right now since we don't copy the traced commands as a buffer into DRAM we do need it.